sqlglot/dialects/duckdb.py

```diff
@@ -2232,3 +2233,64 @@ def bitwisenot_sql(self, expression: exp.BitwiseNot) -> str:
             result_sql = f"~{self.sql(expression, 'this')}"
 
             return _gen_with_cast_to_blob(self, expression, result_sql)
+
+        def window_sql(self, expression: exp.Window) -> str:
+            if isinstance(expression.this, exp.Corr) or (
+                isinstance(expression.this, exp.Filter)
+                and isinstance(expression.this.this, exp.Corr)
+            ):
+                return self.corr_sql(expression)
+
+            return super().window_sql(expression)
+
+        def filter_sql(self, expression: exp.Filter) -> str:
+            if isinstance(expression.this, exp.Corr):
+                return self.corr_sql(expression)
+
+            return super().filter_sql(expression)
+
+        def corr_sql(
+            self: DuckDB.Generator,
+            expression: t.Union[exp.Filter, exp.Window, exp.Corr],
+        ) -> str:
```

Comment:
Ah, I see. You're reusing this for `Filter`, `Window` and `Corr`, hence the transforms entry. I suggest we rename this to `_corr_sql` then to make it clear that it's not intended only for `Corr`. There's an implicit assumption that `<expr>_sql` generates only `Expr`.